### DISCUSSION TITLE
Auto pilot - UE Reference Application UE5

### DISCUSSION THREAD

**Discussion by MariaALJ** [JSBSim-Team/jsbsim#741] [Q&A]
Hello, I recently found about JSBSim during my research about flight dynamics models. Because I'm new to this field, I was happy to learn about the UE Reference Application that uses the JSBSim plugin in Unreal Engine. I have the Reference application now succesfuly  functioning on my UE5. But I'm having trouble moving the aircraft in a way that is physically logical, so my question: is there any flight control commands/autopilot that can be integrated into this application on UE5? Thank you very much.

**Comment by seanmcleod** [comment]
> is there any flight control commands....

I presume you've looked at the **Flight Commands** that appear in the readme for the Unreal plugin here - https://github.com/JSBSim-Team/jsbsim/tree/master/UnrealEngine#flight-commands

If so can you explain what commands you used and what wasn't "physically logical" about the response?

**Reply by MariaALJ** [reply]
Hello, thank you for your response. Maybe I didn't formulate my request correctly. I'm wondering if there are any guidance laws or physical models implemented (like an autopilot mode) that allow to obtain a realistic aircraft trajectory. Thanks!

**Comment by seanmcleod** [comment]
Just to be clear, manual inputs, i.e. without an auto-pilot or some sort of FBW system will produce "physically logical" and "realistic aircraft trajectories" via JSBSim. 

If you want to either develop or make use of an existing auto-pilot or FBW system then first read through the JSBSim Reference Manual - https://github.com/JSBSim-Team/jsbsim#user-documentation and then look at some of the existing aircraft and systems in the JSBSim repo which implement auto-pilots and FBW systems.
